### PR TITLE
Fix ELRS Failsafes at 500Hz and CRSFv3 RX loss on reboot

### PR DESCRIPTION
**Pull request by andyp1per**
Even with RC_OPTIONS = 8192 it was possible to get ELRS failsafes when running at high rates (500Hz).
This PR ensures that rcin ticks run at a fixed period to ensure that we don't miss frames and allows us to lower the CRSF frame timeout meaning we don't miss many multiples of frames in case of failure.

This PR also corrects the reporting of tracer rates and fixes an issues where if CRSFv3 was negotiated the RX could not be reconnected after a soft reboot

Fixes https://github.com/ArduPilot/ardupilot/issues/23853 and https://github.com/ArduPilot/ardupilot/issues/22886

### CODE REVIEW

**Comment by rmackay9** [comment]
@andyp1per,

Txs for this.  Is this related to issue https://github.com/ArduPilot/ardupilot/issues/23853?

**Comment by andyp1per** [comment]
@rmackay9 yes

**Comment by andyp1per** [comment]
This is not yet safe to merge, I am tracking down some watchdogs that might be related

File: libraries/AP_HAL_ChibiOS/Scheduler.cpp [hunk 1]
```diff
@@ -506,8 +506,12 @@ void Scheduler::_rcin_thread(void *arg)
         sched->delay_microseconds(20000);
     }
     while (true) {
-        sched->delay_microseconds(1000);
+        uint32_t now_us = AP_HAL::micros();
         ((RCInput *)hal.rcin)->_timer_tick();
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/Scheduler.cpp:510]: set EK3_MAG_CAL=4 to force mag fusion always, and set GPS_TYPE=14 and do "module load fakegps" in mavproxy. Then wait for the EK3 to say "using GPS" on all EKF lanes, then do the timing
also, separately time the timer_tick with interrupts disabled and look at the timing distribution

File: libraries/AP_RCProtocol/AP_RCProtocol.cpp [hunk 2]
```diff
@@ -295,6 +295,9 @@ static const AP_RCProtocol::SerialConfig serial_configs[] {
 #endif
     // CrossFire:
     { 416666,  0,   1, false },
+    // CRSFv3 can negotiate higher rates which are sticky on soft reboot
```
Comment (tridge) [inline libraries/AP_RCProtocol/AP_RCProtocol.cpp:298]: should check AP_RCPROTOCOL_ELRS_ENABLED?

File: libraries/AP_RCProtocol/AP_RCProtocol.cpp [hunk 3]
```diff
@@ -295,6 +295,9 @@ static const AP_RCProtocol::SerialConfig serial_configs[] {
...
+    { 1000000, 0,   1, false },
```
Comment (tridge) [inline libraries/AP_RCProtocol/AP_RCProtocol.cpp:299]: 1M not needed?

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by andyp1per** [comment]
Timer tick without interrupts, 500Hz ELRS:

image

So about 150-160us when a frame is processed (much quicker when not a complete frame, but most of the time there are complete frames). 7us for incomplete frames. Which is a fairly significant chunk of the 1Khz cycle.

**Comment by andyp1per** [comment]
With mag fusion and GPS enabled there is much more variability:

image

but longest delay is about 1.5ms which is still ok for a 500Mhz frame rate.

**Comment by andyp1per** [comment]
Without the change the average frame rate is 800Hz getting down to 500Hz (the danger zone):

image

**Comment by andyp1per** [comment]
I made some changes to take into account the previous cycle - this gives close to the desired 1Khz rate, even with fakegps and magfusion:

image

File: libraries/AP_HAL_ChibiOS/Scheduler.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/Scheduler.cpp]: Done

File: libraries/AP_RCProtocol/AP_RCProtocol.cpp (hunk 3 above)
Comment (andyp1per) [reply libraries/AP_RCProtocol/AP_RCProtocol.cpp]: Removed

File: libraries/AP_RCProtocol/AP_RCProtocol.cpp (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_RCProtocol/AP_RCProtocol.cpp:299]: Fixed

**Comment by andyp1per** [comment]
Flew 5 packs on this today with tracer - all good. Worked perfectly.

**Comment by tridge** [comment]
we discussed changing the uart interface to have a set_frame_mode() call, which sets up a set of frames to receive in, and the uart driver will use the idle interrupt to detect frame start and move to the next frame

**Comment by andyp1per** [comment]
Fixed another way

**Pull request closed without merging**